datafusion/row/src/writer.rs

```diff
@@ -349,7 +349,7 @@ pub(crate) fn write_field_utf8(
     let from = from.as_any().downcast_ref::<StringArray>().unwrap();
     let s = from.value(row_idx);
     let new_width = to.current_width() + s.as_bytes().len();
-    if new_width > to.data.capacity() {
+    if new_width > to.data.len() {
```

Comment:
Do we need to change the  `write_field_utf8` like this?

Reply:
In my opinion, this fix is not right.
The `data` is used to buffer fixed size of layout and row data, if the new width or new length is less than the capacity of data, why need to expand the space.

Reply:
>  if the new width or new length is less than the capacity of data, why need to expand the space.

I think it's because `Vec` in rust doesn't own the space between (len, capacity].

Reply:
thanks for your explain.
Please ignore my comments. @comphead